catalyst.html

```diff
@@ -142,369 +142,410 @@ <h1>Topology ToolKit</h1>
+on your system:
+
+<pre><code>$ make -j&lt;N&gt;</code></pre>
+
+Once the build is finished, enter the following command to install
+your build of ParaView on your system:
+
+<pre><code>$ sudo make install</code></pre>
+
+<b>b) ParaView</b><br>
+
+To configure ParaView's build and generate build commands, enter the
+following commands:
+
+<pre><code>
+$ cd ~/ttk/ParaView-v5.6.0/
+$ mkdir build
+$ cd build/
+$ cmake                                                           \
+      -DCMAKE_BUILD_TYPE=Release                                  \
+      -DPARAVIEW_INSTALL_DEVELOPMENT_FILES=ON                     \
+      -DPARAVIEW_ENABLE_PYTHON=ON                                 \
+      -DVTK_PYTHON_VERSION=3                                      \
+      -DPARAVIEW_BUILD_QT_GUI=OFF                                 \
+      -DPARAVIEW_USE_MPI=ON                                       \
+      -DPARAVIEW_ENABLE_CATALYST=ON                               \
+      -DVTK_USE_X=OFF                                             \
+      -DVTK_OPENGL_HAS_OSMESA=ON                                  \
+      -DVTK_USE_OFFSCREEN=OFF                                     \
+      -DOSMESA_INCLUDE_DIR=/usr/local/include                     \
+      -DOSMESA_LIBRARY=/usr/local/lib/libOSMesa.so                \
+      -DPYTHON_INCLUDE_DIR=/usr/include/python3.6                 \
+      -DPYTHON_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.6m.so \
+      ..
+</code></pre>
+
+The previous <code>cmake</code> options are either required by TTK or
+used to enable in-situ computation for ParaView. (for further details,
+see this <a target="new"
+href="https://www.paraview.org/Wiki/ParaView/ParaView_And_Mesa_3D#Configuring_ParaView_for_use_with_OSMesa">documentation</a>).
+
+Now you can start the compilation process by entering the following
+command, where <code>&lt;N&gt;</code> is the number of available cores
+on your system (this will take a LONG time):
+
+<pre><code>$ make -j&lt;N&gt;</code></pre>
+
+Once the build is finished, enter the following command to install
+your build of ParaView on your system:
+
+<pre><code>$ sudo make install</code></pre>
 
 <b><a name="catalyst_ttk_cmake">c) TTK</a></b><br>
-To enter the configuration menu of TTK's build, enter the following commands:<br>
-<br>
 
-<code>$ cd ~/ttk/ttk-0.9.2/</code><br>
-<code>$ mkdir build</code><br>
-<code>$ cd build/</code><br>
-<code>$ ccmake ../</code><br>
-<br>
+To enter the configuration menu of TTK's build, enter the following
+commands:
 
-The configuration interface opens. Press <code>[c]</code> to configure. At this stage, under Linux,
-TTK's build should be automatically configured correctly by default so you can press <code>[g]</code> to
-generate.<br>
-<br>
+<pre><code>
+$ cd ~/ttk/ttk-0.9.7/
+$ mkdir build
+$ cd build/
+$ cmake                                                     \
+      -DCMAKE_PREFIX_PATH=/usr/local/lib/cmake/paraview-5.6 \
```

Comment:
strange. cmake's finding that path correctly on all installs I experimented (gentoo, ubuntu)

Reply:
I retried yesterday with a new Ubuntu Desktop VM, and TTK CMake configuration script would still fail at https://github.com/topology-tool-kit/ttk/blob/v0.9.7/core/vtk/CMakeLists.txt#L49.

I think CMake overlooks the /usr/local/lib/cmake/paraview-5.6 folder when searching for VTKConfig.cmake.

This bug seems to be resolved in master.

Reply:
my bad, I forgot you were using paraview's internal vtk for that (and not a system wide install of vtk).
makes sense then.